dvc/remote/ssh.py

```diff
@@ -284,107 +305,86 @@ def download(
 
-        ssh = self.ssh(
-            host=to_infos[0]["host"],
-            user=to_infos[0]["user"],
-            port=to_infos[0]["port"],
-        )
-        sftp = ssh.open_sftp()
-
-        for from_info, to_info, name in zip(from_infos, to_infos, names):
-            if to_info["scheme"] != "ssh":
-                raise NotImplementedError
+        with self.sftp(**to_infos[0]) as sftp:
+            for from_info, to_info, name in zip(from_infos, to_infos, names):
+                if to_info["scheme"] != self.scheme:
+                    raise NotImplementedError
 
-            if from_info["scheme"] != "local":
-                raise NotImplementedError
+                if from_info["scheme"] != "local":
+                    raise NotImplementedError
 
-            logger.debug(
-                "Uploading '{}' to '{}/{}'".format(
-                    from_info["path"], to_info["host"], to_info["path"]
+                logger.debug(
+                    "Uploading '{src}' to '{host}/{path}'".format(
+                        from_info["path"], **to_info["host"]
+                    )
                 )
-            )
 
-            if not name:
-                name = os.path.basename(from_info["path"])
+                if not name:
+                    name = os.path.basename(from_info["path"])
 
-            dname = posixpath.dirname(to_info["path"])
-            self._exec(ssh, "mkdir -p {}".format(dname))
+                src = from_info["path"]
+                dest = to_info["path"]
 
-            try:
-                sftp.put(
-                    from_info["path"],
-                    to_info["path"],
-                    callback=create_cb(name),
-                )
-            except Exception:
-                msg = "failed to upload '{}' to '{}/{}'"
-                logger.error(
-                    msg.format(
-                        from_info["path"], to_info["host"], to_info["path"]
-                    )
-                )
-                continue
+                self._mkdir_p(sftp, posixpath.dirname(dest))
 
-            progress.finish_target(name)
-
-        sftp.close()
-        ssh.close()
+                try:
+                    sftp.put(src, dest, callback=create_cb(name))
+                    progress.finish_target(name)
+                except Exception:
+                    logger.error(
+                        "failed to upload '{host}/{path}' to '{dest}'".format(
+                            **from_info, dest=dest
+                        )
+                    )
+                    continue
 
     def list_cache_paths(self):
-        ssh = self.ssh(host=self.host, user=self.user, port=self.port)
-        cmd = "find {} -type f -follow -print".format(self.prefix)
-        stdout = self._exec(ssh, cmd)
-        flist = stdout.split()
-        ssh.close()
+        with self.ssh(**self.path_info) as ssh:
+            cmd = "find {} -type f -follow -print".format(self.prefix)
```

Comment:
list_cache_paths is used for `dvc status -c`, so we also need to re-implement this function without using `find` command.